src/main/java/seedu/edrecord/model/ModelManager.java

```diff
@@ -178,7 +178,10 @@ public void addModule(Module mod) {
     public void setModuleFilter(PartOfModulePredicate modulePredicate) {
         requireNonNull(modulePredicate);
         this.selectedModulePredicate = modulePredicate;
-        this.selectedModule = modulePredicate.getModule();
+
+        String currentModuleCode = modulePredicate.getModuleCode();
+        this.selectedModule = moduleSystem.getModule(currentModuleCode);
```

Comment:
Same as above, what's the rationale behind changing to module code for the predicate instead?